qt/widgets/plugins/algorithm_dialogs/src/LoadDialog.cpp

```diff
@@ -111,29 +111,32 @@ void LoadDialog::enableNameSuggestion(const bool on) {
   }
 }
 
+void LoadDialog::enableLoadRequests() { m_okButton->blockSignals(false); }
+
+void LoadDialog::disableLoadRequests() { m_okButton->blockSignals(true); }
+
 /**
  * Called when the run button is clicked
  */
 void LoadDialog::accept() {
   // The file widget may have been edited but not lost focus so that the search
-  // wasn't
-  // attempted for the new contents. Force one here.
+  // wasn't attempted for the new contents. Force one here.
   // The widget does nothing if the contents have not changed so it will be
   // quick for this case
+  disableLoadRequests();
```

Comment:
I'm wondering if something that might throw an exception in the middle of this and `enable` could leave the button disabled? For memory allocation we'd use the RAII pattern, how about using the same thing here to ensure that any route out leaves the button enabled again?